Sources/StreamChat/WebSocketClient/EventMiddlewares/UserWatchingEventMiddleware.swift

```diff
@@ -16,19 +16,35 @@ struct UserWatchingEventMiddleware<ExtraData: ExtraDataTypes>: EventMiddleware {
         
         database.write { session in
             guard let channelDTO = session.channel(cid: userWatchingEvent.cid) else {
-                throw ClientError.ChannelDoesNotExist(cid: userWatchingEvent.cid)
+                let currentUserId = session.currentUser()?.user.id
+                if userWatchingEvent.userId == currentUserId {
+                    log.info(
+                        "Ignoring watcher event for channel \(userWatchingEvent.cid) and current user"
+                            + "since Channel doesn't exist locally."
+                    )
+                } else {
+                    log.error(
+                        "Failed to save watcher event for channel \(userWatchingEvent.cid)"
+                            + "and user \(userWatchingEvent.userId) since Channel doesn't exist locally."
+                    )
+                }
+                return
             }
             
-            guard let userDTO = session.user(id: userWatchingEvent.userId) else {
-                throw ClientError.UserDoesNotExist(userId: userWatchingEvent.userId)
-            }
+            channelDTO.watcherCount = Int64(userWatchingEvent.watcherCount)
             
-            if userWatchingEvent.isStarted {
-                channelDTO.watchers.insert(userDTO)
+            if let eventPayload = userWatchingEvent.payload as? EventPayload<ExtraData>,
+               let userPayload = eventPayload.user {
+                let userDTO = try session.saveUser(payload: userPayload)
```

Comment:
Isn't the payload already saved in `EventDataProcessorMiddleware`? IMO the original version of this was completely fine.